### ISSUE TITLE
Current implementation of images inspect is faulty (eg: RepoTags)

### ISSUE THREAD

**Issue by apostasie** [containerd/nerdctl#3004]
### Description

nerdctl does not deal properly with images bounds to multiple tags.

### Steps to reproduce the issue

Compare:
```
sudo ./nerdctl pull dubodubonduponey/testouille
sudo ./nerdctl pull dubodubonduponey/testouille:2
sudo ./nerdctl inspect dubodubonduponey/testouille
sudo ./nerdctl inspect XXXXXXX
```

with

```
docker pull dubodubonduponey/testouille
docker pull dubodubonduponey/testouille:2
docker inspect dubodubonduponey/testouille
docker inspect XXXXXXX
```

### Describe the results you received and expected

1. when querying by reference, nerdctl returned result has in RepoTags ONLY the name used to query, regardless of the fact other tags may hold the same image
2. when querying by id (that is bound to multiple tags), nerdctl returns multiple distinct images instead of just one

Compare with docker behavior.

### What version of nerdctl are you using?

1.7.6

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

**Comment by apostasie** [comment]
Note that the implementation is https://github.com/containerd/nerdctl/blob/4fff6b22b3efab115a7727394a69f5251d2564cb/pkg/cmd/image/inspect.go#L34
is problematic from the get-go, not just the implementation of `ImageFromNative`.

It natively assumes that every successive image returned from the walker needs a different entry.

**Comment by apostasie** [comment]
Fixed by #3017
